browser/resources/settings/email_aliases_page/content/email_aliases_modal.tsx

```diff
@@ -0,0 +1,182 @@
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Button from "@brave/leo/react/button"
+import Icon from "@brave/leo/react/icon"
+import { getLocale } from '$web-common/locale'
+import formatMessage from '$web-common/formatMessage'
+import * as React from 'react'
+import { ViewState, ViewMode, MappingService, MAX_ALIASES } from "./types"
+import Col from "./styles/Col"
+import Row from "./styles/Row"
+import Input from "@brave/leo/react/input"
+import styled from "styled-components"
+import onEnterKey from "./onEnterKey"
+import { color, font, spacing } from "@brave/leo/tokens/css/variables"
+
+
+const ModalSectionCol = styled(Col)`
+  margin: ${spacing["2Xl"]} 0;
+  & h3 {
+    margin: ${spacing.s} ${spacing.m};
+  }
+  & leo-input {
+    margin: ${spacing.s} 0;
+  }
+`
+
+const ButtonRow = styled(Row)<{ bubble: boolean }>`
+  justify-content: ${props => props.bubble ? 'space-between' : 'end'};
+  margin: ${spacing["2Xl"]} 0;
+  & leo-button {
+    flex-grow: 0;
+  }
+  & span {
+    display: flex;
+    flex-direction: row;
+    column-gap: ${spacing.m};
+  }
+`
+
+const GeneratedEmailContainer = styled(Row)`
+  font: ${font.large.regular};
+  background-color: ${color.neutralVariant[10]};
+  border-radius: ${spacing.m};
+  padding: 0 0 0 ${spacing.m};
+  margin: ${spacing.s} 0;
+  justify-content: space-between;
+  height: ${spacing["5Xl"]};
+  & leo-button {
+    flex-grow: 0;
+  }
+`
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  @keyframes spin {
+    0% { transform: rotate(0deg); }
+    100% { transform: rotate(360deg); }
+  }
+  & .waiting {
+    animation: spin 1s linear infinite;
+  }
+  width: ${spacing["4Xl"]};
+  height: 100%;
+`
+
+const RefreshButton = ({ onClick, waiting }: { onClick: () => Promise<void>, waiting: boolean }) => {
+  return <ButtonWrapper title={getLocale('emailAliasesRefreshButtonTitle')}>
+    {waiting
+      ? <Icon className='waiting' name="loading-spinner" />
+      : <Button title={getLocale('emailAliasesGeneratingNewAlias')}
+        onClick={onClick}
+        kind="plain" >
+        <Icon name="refresh" />
+      </Button>}
+  </ButtonWrapper>
+}
+
+export const EmailAliasModal = (
+  { onReturnToMain, viewState, email, mode, mappingService, bubble }:
+    {
+      onReturnToMain: () => void,
+      viewState?: ViewState,
+      bubble?: boolean,
+      mode: ViewMode,
+      email: string,
+      mappingService: MappingService
+    }
+) => {
+  const [limitReached, setLimitReached] = React.useState<boolean>(false)
+  const [mainEmail, setMainEmail] = React.useState<string>(email)
+  const [proposedAlias, setProposedAlias] = React.useState<string>(viewState?.alias?.email ?? '')
+  const [proposedNote, setProposedNote] = React.useState<string>(viewState?.alias?.note ?? '')
+  const [awaitingProposedAlias, setAwaitingProposedAlias] = React.useState<boolean>(true)
```

Comment:
I think this state could be pushed into refreshButton - this component doesn't actually care about it

Reply:
I think this state is needed in the EmailAliasModal component because of the state of the "create" button.

Reply:
Whoops missded that!